builtin_array.go

```diff
@@ -173,7 +173,7 @@ func (r *Runtime) arrayproto_pop(call FunctionCall) Value {
 
 func (r *Runtime) arrayproto_join(call FunctionCall) Value {
 	o := call.This.ToObject(r)
-	o.AddSeenObject(o)
+	o.addSeenObject(o)
```

Comment:
[nit] how do you feel about moving the delete up here with a defer to keep these two lines close?

```
o.addSeenObject(o)
defer o.removeSeenObject(o) // remove when done to allow printing multiple times
```